crates/rustc_codegen_spirv/src/linker/inline.rs

```diff
@@ -13,124 +13,158 @@ use rustc_data_structures::fx::{FxHashMap, FxHashSet};
 use rustc_session::Session;
 use std::mem::take;
 
-type FunctionMap = FxHashMap<Word, Function>;
+type FunctionMap = FxHashMap<Word, usize>;
 
 pub fn inline(sess: &Session, module: &mut Module) -> super::Result<()> {
+    let (disallowed_argument_types, disallowed_return_types) =
+        compute_disallowed_argument_and_return_types(module);
+    let mut to_delete: Vec<_> = module
+        .functions
+        .iter()
+        .map(|f| should_inline(&disallowed_argument_types, &disallowed_return_types, f))
+        .collect();
     // This algorithm gets real sad if there's recursion - but, good news, SPIR-V bans recursion
```

Comment:
Making this comment a bit more explicit instead of referring to the extremely tiny and easy to miss `?` would be nice.